### PR TITLE
Rename NetLB forwarding rule

### PR DESCRIPTION
**Pull request by kl52752**
Forwarding rule name for NetLB should align with legacy forwarding rule
name.

### CODE REVIEW

File: pkg/loadbalancers/l4netlb.go [hunk 1]
```diff
@@ -298,22 +286,3 @@ func (l4netlb *L4NetLB) createFirewalls(name, hcLink, hcFwName string, hcPort in
 	}
 	return string(protocol), result
 }
-
-func (l4netlb *L4NetLB) cleanupForwardingRuleIfProtocolChanged(name, protocol string) (*composite.ForwardingRule, error) {
```
Comment (cezarygerard) [inline pkg/loadbalancers/l4netlb.go:302]: why did you delete this?

File: pkg/loadbalancers/forwarding_rules.go [hunk 2]
```diff
@@ -321,32 +321,33 @@ func (l *L4) deleteForwardingRule(name string, version meta.Version) {
...
 	ipToUse := l4lbIPToUse(l4netlb.Service, existingFwdRule, "")
-	klog.V(2).Infof("ensureExternalForwardingRule(%v): LoadBalancer IP %s", loadBalancerName, ipToUse)
+	klog.V(2).Infof("ensureExternalForwardingRule(%v): LoadBalancer IP %s", frName, ipToUse)
 
 	var addrMgr *addressManager
 	// If the network is not a legacy network, use the address manager
 	if !l4netlb.cloud.IsLegacyNetwork() {
 		nm := types.NamespacedName{Namespace: l4netlb.Service.Namespace, Name: l4netlb.Service.Name}.String()
-		addrMgr = newAddressManager(l4netlb.cloud, nm, l4netlb.cloud.Region(), "", loadBalancerName, ipToUse, cloud.SchemeExternal)
+		addrMgr = newAddressManager(l4netlb.cloud, nm, l4netlb.cloud.Region(), "", frName, ipToUse, cloud.SchemeExternal)
```
Comment (cezarygerard) [inline pkg/loadbalancers/forwarding_rules.go:345]: the empty string arg could ne replaced with const

noSubnetUrl = ""

File: pkg/loadbalancers/l4netlb.go (hunk 1 above)
Comment (kl52752) [reply pkg/loadbalancers/l4netlb.go:302]: Because using namer for forwarding rule we add protocol to name so if protocol changed we need to delete the old fire wall because its name will change. Now when we use legacy name we don't change name for forwarding rule so we can just update it.

File: pkg/loadbalancers/forwarding_rules.go (hunk 2 above)
Comment (kl52752) [reply pkg/loadbalancers/forwarding_rules.go]: done

File: pkg/loadbalancers/forwarding_rules.go [hunk 3]
```diff
@@ -321,32 +321,34 @@ func (l *L4) deleteForwardingRule(name string, version meta.Version) {
...
 	ipToUse := l4lbIPToUse(l4netlb.Service, existingFwdRule, "")
-	klog.V(2).Infof("ensureExternalForwardingRule(%v): LoadBalancer IP %s", loadBalancerName, ipToUse)
+	klog.V(2).Infof("ensureExternalForwardingRule(%v): LoadBalancer IP %s", frName, ipToUse)
 
 	var addrMgr *addressManager
 	// If the network is not a legacy network, use the address manager
 	if !l4netlb.cloud.IsLegacyNetwork() {
 		nm := types.NamespacedName{Namespace: l4netlb.Service.Namespace, Name: l4netlb.Service.Name}.String()
-		addrMgr = newAddressManager(l4netlb.cloud, nm, l4netlb.cloud.Region(), "", loadBalancerName, ipToUse, cloud.SchemeExternal)
+		noSubnetUrl := ""
```
Comment (freehan) [inline pkg/loadbalancers/forwarding_rules.go:345]: nit: no need for this variable. If you want comment, you can add it next to the empty string parameter.

**Review by freehan — COMMENTED** [review]
just nit

LGTM overall

File: pkg/loadbalancers/forwarding_rules.go [hunk 4]
```diff
@@ -321,32 +321,34 @@ func (l *L4) deleteForwardingRule(name string, version meta.Version) {
...
 	// version used for creating the existing forwarding rule.
 	version := meta.VersionGA
+	existingFwdRule := l4netlb.GetForwardingRule(frName, version)
 
 	// Determine IP which will be used for this LB. If no forwarding rule has been established
 	// or specified in the Service spec, then requestedIP = "".
 	// TODO(kl52752) Add support for NetworkTier
 	ipToUse := l4lbIPToUse(l4netlb.Service, existingFwdRule, "")
-	klog.V(2).Infof("ensureExternalForwardingRule(%v): LoadBalancer IP %s", loadBalancerName, ipToUse)
+	klog.V(2).Infof("ensureExternalForwardingRule(%v): LoadBalancer IP %s", frName, ipToUse)
```
Comment (prameshj) [inline pkg/loadbalancers/forwarding_rules.go:339]: nit: you can use %s for frName as well.

File: pkg/loadbalancers/l4netlb.go [hunk 5]
```diff
@@ -251,12 +244,7 @@ func (l4netlb *L4NetLB) EnsureLoadBalancerDeleted(svc *corev1.Service) *SyncResu
 // GetFRName returns the name of the forwarding rule for the given L4 External LoadBalancer service.
 // This appends the protocol to the forwarding rule name, which will help supporting multiple protocols in the same service.
```
Comment (prameshj) [inline pkg/loadbalancers/l4netlb.go:245]: Fix this comment and also explain why we are using the legacy naming scheme.

File: pkg/loadbalancers/forwarding_rules.go (hunk 3 above)
Comment (kl52752) [reply pkg/loadbalancers/forwarding_rules.go:345]: done

File: pkg/loadbalancers/forwarding_rules.go (hunk 4 above)
Comment (kl52752) [reply pkg/loadbalancers/forwarding_rules.go:339]: done

File: pkg/loadbalancers/l4netlb.go (hunk 5 above)
Comment (kl52752) [reply pkg/loadbalancers/l4netlb.go:245]: done

**Review by freehan — APPROVED** [review]
/lgtm
/approve
